### DISCUSSION TITLE
how to apply a if condition for the name of the branch we get using list of branches api in gitactions

### DISCUSSION THREAD

**Discussion by j-jagan** [community/community#40428] [Apps, API and Webhooks]
### Select Topic Area

Show & Tell

### Body

When we run this api for getting the list of branches for a repository in gitactions workflow:

curl \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer <YOUR-TOKEN>" \
  https://api.github.com/repos/OWNER/REPO/branches

It gives the result in the given format:

[
  {
    "name": "master",
    "commit": {
      "sha": "c5b97d5ae6c19d5c5df71a34c7fbeeda2479ccbc",
      "url": "https://api.github.com/repos/octocat/Hello-World/commits/c5b97d5ae6c19d5c5df71a34c7fbeeda2479ccbc"
    },
    "protected": true,
    "protection": {
      "required_status_checks": {
        "enforcement_level": "non_admins",
        "contexts": [
          "ci-test",
          "linter"
        ]
      }
    },
    "protection_url": "https://api.github.com/repos/octocat/hello-world/branches/master/protection"
  }
]

so, from the result, we are getting the name of the branch , how to store that branch name in a variable in our workflow.

**Comment by airtower-luna** [comment]
In principle, you need to parse the value you want out of the JSON response. Then you can [store it in an environment variable or output](https://github.com/actions/checkout#push-a-commit-using-the-built-in-token).

The minimal way to do JSON parsing is probably with [`jq`](https://github.com/stedolan/jq), e.g.:

```
curl -H "Accept: application/vnd.github+json" -H "Authorization: Bearer " https://api.github.com/repos/OWNER/REPO/branches | jq -r .[0].name
```

However, mind that the response contains a list of branches. The `[0]` in the example just selects the very first element. You might need some kind of mechanism to find the branch you want to look at. Of course you could use any other kind of JSON parser or library, depending on what works best for what you want to do.
